### DISCUSSION TITLE
Erro no código JavaScript

### DISCUSSION THREAD

**Discussion by lnr2289** [community/community#72635] [Programming Help]
### Body

Podem me ajudar? Como encontro o erro nesse código em JS?
<!DOCTYPE html>
<html lang="pt-br">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Gerador de Tabela de Jogos</title>
</head>

<body>
    <form>
        <h1>Campeonato Eliminatório</h1>
        <p>
        <h2>Adicionar Clube</h2>
        <input type="text" id="clube" placeholder="Nome do clube">
        <button onclick="adicionarClube()">Adicionar Clube</button>
        </p>
        <h2>Lista de Clubes</h2>
        <ul id="lista-de-clubes"></ul>
        <button onclick="ListaClubes()">Lista de Clubes</button>
        <p>
        <h2>Tabela de Jogos</h2>
        <button onclick="TabelaDeJogos()">Gerar Tabela de Jogos</button>
        <table id="tabela-de-jogos"></table>
        </p>
    </form>
    <pre></pre>
    <script src="ex6_3.js"></script>
</body>

</html>

Código JS

const frm = document.querySelector("form") // obtém os elementos da página
var clubes = [];

function adicionarClube() {
    var clube = document.querySelector("clube").value;
    if (clube.trim() !== "") {
        clubes.push(clube);
        document.querySelector("clube").value = "";
        listarClubes();
    } else {
        alert("Informe o nome do clube.");
    }
}
function listarClubes() {
    var listaDeClubes = document.querySelector("lista-de-clubes");
    listaDeClubes.innerHTML = "";
    for (var i = 0; i < clubes.length; i++) {
        var li = document.createElement("li");
        li.appendChild(document.createTextNode(clubes[i]));
        listaDeClubes.appendChild(li);
    }
}
function TabelaDeJogos() {
    if (clubes.length % 2 !== 0) {
        alert("O número de clubes deve ser par para formar uma tabela de jogos.");
        return;
    }
    var tabelaDeJogos = document.querySelector("tabela-de-jogos");
    tabelaDeJogos.innerHTML = "";
    var rodada = 1;
    for (var i = 0; i < clubes.length / 2; i++) {
        var jogo = clubes[i] + " x " + clubes[clubes.length - 1 - i];
        var tr = document.createElement("tr");
        var td = document.createElement("td");
        td.appendChild(document.createTextNode("Rodada " + rodada + ": " + jogo));
        tr.appendChild(td);
        tabelaDeJogos.appendChild(tr);
        rodada++;
    }
}

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Bashamega** [comment]
Hello Luciano :)
I have found multiple mistakes in the js code.
These are the problems:

- Incorrect Selector: The original code used an incorrect selector to query the HTML elements. For example, it used "club" to select an element with the ID of "club," which should be #club.

- Variable Case: The variable names were inconsistent and had both uppercase and lowercase letters. It's good practice to use consistent variable naming conventions.

- Undefined Variables: In the `listClubs` and `generateGameTable` functions, the code used variables like `Clublist`, Round, and Game that were not defined. These should be lowercase.

- Typo: The `functionGameTable` was written as `functionGameTable`, which is not a valid function declaration.

- Function Naming: The `generateGameTable` function was named correctly in the corrected code to follow JavaScript naming conventions.

- Missing return;: In the `generateGameTable` function, there was an alert message for an odd number of clubs, but the function continued to execute. In the corrected code, it was modified to include a return; statement to exit the function.

- HTML Element Selection: The `querySelector` used to select HTML elements was missing the # for IDs, which is the correct way to select elements by ID.

**Comment by lnr2289** [comment]
I understand, Basha. I'm just getting started in the field of programming, and I can see that I have a lot of progress to make. How can I improve my programming logic? Do you have any tips for me?

**Comment by Bashamega** [comment]
> How can I improve my programming logic?

There isn't lot to it.
Just keep practicing and trying to solve an bugs you have and maybe try reading other peoples code to get a background of how people write code, and the most important thing is to be positive and fearless.

**Reply by lnr2289** [reply]
"I have been improving my practice, but I see that there is still much to add. I will follow your guidance. In which programming language do you code?"

**Reply by Bashamega** [reply]
> "I have been improving my practice, but I see that there is still much to add. I will follow your guidance. In which programming language do you code?"

I code in multiple languages.
HTML, CSS, JS, PY, tailwind, and some js frameworks. and I am currently learning dart

**Reply by lnr2289** [reply]
"Everything is practice combined with theory, right, Basha? I'll keep practicing and reading other people's code to try and improve my own. Can you help me review some of the code I'm creating if it doesn't bother you, of course?"

**Reply by Bashamega** [reply]
Ofcourse I don't bother.
